tests/utils/test_run_params.py

```
@@ -0,0 +1,338 @@
+    params_path = tmp_path / "params.json"
+    params_path.write_text(json.dumps({"model": "boltz2", "models": ["rf3"]}))
+
+    with pytest.raises(ValueError, match="conflicting model"):
+        apply_run_params(_namespace(params=str(params_path)))
+
+
+def test_apply_run_params_rejects_missing_inputs(tmp_path):
+    params_path = tmp_path / "params.json"
+    params_path.write_text(json.dumps({"model": "boltz2"}))
+
+    with pytest.raises(ValueError, match="inputs.proteins or proteins_path"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)